browser/brave_rewards/rewards_service_factory.cc

```diff
@@ -65,6 +66,10 @@ RewardsService* RewardsServiceFactory::GetForProfile(
     return testing_service_;
   }
 
+  if (profile->IsOffTheRecord() || brave::IsTorProfile(profile)) {
```

Comment:
@NejcZdovc Is this correct that for testing the AdsService is associated to all profiles?

Reply:
Maybe we also need to check about guest profile?

Reply:
we've actually be discussing something like profile::IsRegularProfile because these kinds of checks are very fragile cc @yrliou who is working on something

Reply:
+1 for `profile:IsRegularProfile`!

Reply:
Current plan https://github.com/brave/brave-browser/issues/11218#issuecomment-673767926, will do it in another PR.

Reply:
Closing this issue as https://github.com/brave/brave-browser/issues/11218 will resolve the issue for all services including BAT Ads and BAT Ledger (Brave Rewards)

Reply:
@simonhong From comments in profile.h, IsOffTheRecord is true for guest sessions which includes guest profile.

Reply:
@yrliou IIRC, When guest window is created, one normal profile is created and it acts as a parent profile of otr profile that is effective profile for gest window. I checked before that new rewards service is created when guest window is newly launched.

Reply:
@simonhong Yeah, I guess chromium's guest window has the same behavior as our Tor profile in this case, we both create a normal profile as the parent profile and their IsOffTheRecord will return false. Their comments in profile.h is a bit confusing, thanks for pointing out your test result for guest windows here.